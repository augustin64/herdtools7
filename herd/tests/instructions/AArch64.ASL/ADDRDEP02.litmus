AArch64 ADDRDEP02
(* Test addr;po in lob, addr to store *)
{
0:X0=x; 0:X3=y;
1:X0=y; 1:X4=z; 1:X6=x;
}
 P0           | P1                  ;
 LDR W1,[X0]  | LDR W1,[X0]         ;
 MOV W2,#1    | EOR W2,W1,W1        ;
 STLR W2,[X3] | MOV W3,#1           ;
              | STR W3,[X4,W2,SXTW] ;
              | MOV W5,#1           ;
              | STR W5,[X6]         ;
~exists (0:X1=1 /\ 1:X1=1)
