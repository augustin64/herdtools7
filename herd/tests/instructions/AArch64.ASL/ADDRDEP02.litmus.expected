Test ADDRDEP02 Forbidden
States 3
0:X1=0; 1:X1=0;
0:X1=0; 1:X1=1;
0:X1=1; 1:X1=0;
Ok
Witnesses
Positive: 3 Negative: 0
Condition ~exists (0:X1=1 /\ 1:X1=1)
Observation ADDRDEP02 Never 0 3
Hash=2c5ea9476c1c4a84f083fea413b7c4a8

